api/src/test/java/org/asynchttpclient/test/TestUtils.java

```diff
@@ -92,6 +99,57 @@ public static File createTempFile(int approxSize) throws IOException {
         }
     }
 
+    public static class SyncByteBufferPublisher implements Publisher<ByteBuffer> {
+        // FIXME If I reduce the chunk size the test will fail. Why?
+        private static final int CHUNK_SIZE = 30000;
+
+        private final byte[] bytes; // payload streamed to the subscriber
+        private final int chunkSize; // number of bytes in each element sent to
+                                     // the subscriber
+
+        public SyncByteBufferPublisher(byte[] bytes) {
+            this(bytes, CHUNK_SIZE);
+        }
+
+        public SyncByteBufferPublisher(byte[] bytes, int chunkSize) {
+            this.bytes = Arrays.copyOf(bytes, bytes.length);
+            this.chunkSize = chunkSize;
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super ByteBuffer> subscriber) {
+            subscriber.onSubscribe(new Subscription() {
+                // This index is used to keep track of bytes that have been sent.
+                private volatile int currentIndex = 0;
+                // This flag will track whether this `Subscription` is to be considered cancelled or not
+                private volatile boolean cancelled = false;
+
+                public void request(long n) {
```

Comment:
> If we were to go with Akka, I guess we would have to turn the maven project into a Scala/Java mixed one so we can use TestKit.

We don't need the TestKit. A dependency to akka-streams would be enough. But, even if we had to use TesKit, I don't see why we would then need to turn this project into a Scala/Java mixed one.

That said, I think we should go with rxjava because with akka-streams turning a `Source` into a `Publisher` requires materialization (meaning we need to create an `ActorMaterializer` and hence also an `ActorSystem`). Consider that with rxjava it's straightforward to convert an `Observer` into a `Publisher`.